### PR TITLE
updated to aff 4.x

### PR DESCRIPTION
**Pull request by Thimoteus**
Sprinkled a few `nonCanceler`s in there, not sure if you're OK with that or what.

### CODE REVIEW

**Comment by Thimoteus** [comment]
I'm not sure why travis is failing. The error there looks like what happens if you just update the dependencies and try to run it against unupdated code, but that was already fixed.

`rm -rf output/ bower_components/` then `bower i` and `pulp build` compiles successfully for me.

**Comment by kritzcreek** [comment]
The problem is in https://github.com/kRITZCREEK/purescript-psc-ide/blob/master/src/Node/Which.purs

That probably needs some EffFnAff

EDIT: nevermind, I see the changes you made now

**Comment by kritzcreek** [comment]
I see why it's failing now, the build has been switched to use psc-package. I'm assuming we need to update the package set.

**Comment by Thimoteus** [comment]
Oh, of course. OK, should be fine now.

**Comment by kritzcreek** [comment]
Looking good, thanks :) Looks good to you @nwolverson? Are the plugins still on Aff3?

File: src/Node/Which.purs [hunk 1]
```diff
@@ -16,7 +18,9 @@ foreign import whichImpl :: forall eff.
   Eff (fs :: FS | eff) Unit
 
 which :: forall eff. String -> Aff (fs :: FS | eff) (Array String)
-which = makeAff <<< whichImpl { path: toNullable Nothing, pathExt: toNullable Nothing }
+which s = makeAff \ cb -> nonCanceler <$
```
Comment (nwolverson) [inline src/Node/Which.purs:21]: what's with sometimes `\ cb ->` and sometimes `\cb ->` ?

File: src/PscIde/Server.purs [hunk 2]
```diff
@@ -89,20 +89,20 @@ startServer { stdio, exe, combinedExe, cwd, source, port, directory, outputDirec
...
                       onClose cp (\exit -> case exit of
-                                     (Normally 0) -> succ Closed
-                                     (Normally n) -> succ $ StartError $ "Error code returned: "<> show n
-                                     _ -> succ $ StartError "Other close error")
+                                     (Normally 0) -> cb $ Right Closed
+                                     (Normally n) -> cb $ Right $ StartError $ "Error code returned: "<> show n
+                                     _ -> cb $ Right $ StartError "Other close error")
 
-    unwrap (parallel handleErr <|> parallel (delay (Milliseconds 100.0) $> Started cp))
+    sequential (parallel handleErr <|> parallel (delay (Milliseconds 100.0) $> Started cp))
```
Comment (nwolverson) [inline src/PscIde/Server.purs:105]: `sequential` `parallel`, nice

**Comment by nwolverson** [comment]
I guess so, that's just laziness. 

LGTM

**Pull request merged**